doc/sphinx/language/gallina-extensions.rst

```diff
@@ -768,42 +804,78 @@ Section :ref:`gallina-definitions`).
 .. cmd:: End @ident
 
    This command closes the section named :token:`ident`. After closing of the
-   section, the local declarations (variables and local definitions) get
+   section, the local declarations (variables and local definitions, see :cmd:`Variable`) get
    *discharged*, meaning that they stop being visible and that all global
    objects defined in the section are generalized with respect to the
    variables and local definitions they each depended on in the section.
 
-   .. example::
+   .. exn:: This is not the last opened section.
+      :undocumented:
 
-      .. coqtop:: all
+.. note::
+   Most commands, like :cmd:`Hint`, :cmd:`Notation`, option management, … which
+   appear inside a section are canceled when the section is closed.
+
+.. cmd:: Variable @ident : @type
 
-         Section s1.
+   This command links :token:`type` to the name :token:`ident` in the context of
+   the current section. When the current section is closed, name :token:`ident`
+   will be unknown and every object using this variable will be explicitly
+   parametrized (the variable is *discharged*).
 
-         Variables x y : nat.
+   .. exn:: @ident already exists.
+      :name: @ident already exists. (Variable)
+      :undocumented:
 
-         Let y' := y.
+   .. cmdv:: Variable {+ @ident } : @term
 
-         Definition x' := S x.
+      Links :token:`type` to each :token:`ident`.
```

Comment:
There is an inconsistency between the description and the command. These variants should use the token `type` like the main one.